generic/reg-util.py

```diff
@@ -0,0 +1,240 @@
+        REG_C1          0x00C1  0xC159  LASER_TRANS_6[47:32]
+        REG_C2          0x00C2  0xC25A  LASER_TRANS_6[47:32]
+        REG_C4          0x00C4  0xC45C  LASER_SET_POINT
+        REG_C8          0x00C8  0xC860  COMPILE_OPTIONS[31:16]
+        REG_C9          0x00C9  0xC961  COMPILE_OPTIONS[15:0]
+        REG_CC          0x00CC  0xCC64  LINE_LENGTH (Last Legacy)
+        REG_D0          0x00D0  0xD068  Device ID (RD Only)
+        REG_D4          0x00D4  0xD46C  Status (RD Only)
+        REG_D8          0x00D8  0xD870  Trigger Mode
+        REG_DC          0x00DC  0xDC74  Acquisition Mode
+        REG_DD          0x00DD  0xDD78  Acquisition Delay
+        REG_E0          0x00E0  0xE07C  FPGA Configuration
+        REG_E4          0x00E4  0xE480  ADC Configuration
+        REG_E8          0x00E8  0xE881  FIFO RD Pointer
+        REG_E9          0x00E9  0xE982  FIFO WR Pointer
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        if not self.init_usb():
+            return
+
+        self.init_table()
+        self.init_gui()
+
+    def init_usb(self):
+        self.dev = None
+        for dev in usb.core.find(find_all=True):
+            if dev.idVendor == 0x24aa and dev.idProduct == 0x4000:
+                print(f"found VID 0x{dev.idVendor:04x} PID 0x{dev.idProduct:04x}")
+                self.dev = dev
+                return True
+
+    def init_gui(self):
+        self.title("FPGA Register Utility")
+        self.geometry("400x200")
+
+        row = 0  # [         | Address  |  Value  ]
+        tk.Label(text="Address").grid(row=row, column=1)
+        tk.Label(text="Value").grid(row=row, column=2)
+
+        row += 1 # [ (Write) | [name v] | [_____] ]
+        tk.Button(text="Write", command=self.write_callback).grid(row=row, column=0)
+        self.write_addr = self.make_addr_combobox(row, 1)
+        self.write_value = tk.Entry(width=6)
+        self.write_value.grid(row=row, column=2)
+
+        row += 1 # [ (Read)  | [name v] | [_____] ]
+        tk.Button(text="Read", command=self.read_callback).grid(row=row, column=0)
+        self.read_addr = self.make_addr_combobox(row, 1)
+        self.read_value = tk.StringVar(value="0xBEEF")
+        tk.Label(height=1, width=6, textvariable=self.read_value).grid(row=row, column=2)
+
+        row += 1 # [ (__________READ_ALL________) ]
+        tk.Button(text="Read All", width=30, command=self.read_all_callback).grid(row=row, column=0, columnspan=3)
+
+        # keyboard shortcuts (untested)
+        self.bind('<Control-R>', self.read_callback)
+        self.bind('<Control-W>', self.write_callback)
+        self.bind('<Control-V>', self.write_value.set_focus)
+
+    ############################################################################
+    # event callbacks
+    ############################################################################
+
+    def write_callback(self):
+        """ user clicked the "write" button """
```

Comment:
Would be nice to have a comment providing some context about self.write_addr. 
```
# self.write_addr is a tk.StringVar contained in a ttk.ComboBox
```

It's also helpful to write callbacks using the `sender` parameter